File: 7_kyu/Looking_for_a_benefactor.go
package kata

func NewAvg(arr []float64, navg float64) int64 {
    var delta float64
    for _,v := range arr { delta += navg - v }
    delta += navg
    if delta < 0 {
        return -1
     } else {
         return int64(delta+0.5)
     }
}

File: 7_kyu/Looking_for_a_benefactor.lua
local solution = {}

-- Sum all values in the arr.
function solution.sum(arr)
  local accu = 0
  for _,x in ipairs(arr) do
    accu = accu + x
  end
  return accu
end

-- Returns x that arithmetic mean of arr and x = navg.
-- Returns x rounded to an integer value.
-- Returns -1 if a negative value is needed.
function solution.newAvg(arr, navg)
  local missing = navg * (1+#arr) - solution.sum(arr)
  missing = math.floor(missing+0.5)
  missing = math.max(-1,missing)
  return missing
end

return solution
